libcontainer/cgroups/fs2/devices.go

```diff
@@ -26,11 +28,21 @@ func isRWM(perms devices.Permissions) bool {
 	return r && w && m
 }
 
-// the logic is from crun
-// https://github.com/containers/crun/blob/0.10.2/src/libcrun/cgroup.c#L1644-L1652
 func canSkipEBPFError(cgroup *configs.Cgroup) bool {
+	// If we're running in a user namespace we can ignore eBPF rules because
+	// we're already restricted to not be able to mknod(2) by default.
+	if system.RunningInUserNS() {
+		return true
+	}
+
+	// We cannot ignore an eBPF load error if any rule if is a block rule or it
+	// doesn't permit all access modes.
+	//
+	// NOTE: This will sometimes trigger in cases where access modes are split
+	//       between different rules but to handle this correctly would require
+	//       using ".../libcontainer/cgroup/devices".Emulator.
 	for _, dev := range cgroup.Resources.Devices {
-		if dev.Allow || !isRWM(dev.Permissions) {
+		if !dev.Allow || !isRWM(dev.Permissions) {
```

Comment:
Can we have an integration test for this? Also we need to make sure this is not a vulnerability.

Reply:
If the error path can be hit by non-rootless containers then yeah this is probably a vulnerability.

Reply:
I'll need to think about the easiest way of integration testing this one -- we have to trigger a syscall failure.

Reply:
I tried to do this but it seems like it's not really simple to force a failure with `bpf(2)` -- the obvious way of doing it (making a program that is too big so it trips up the eBPF program size rules) appears to cause the test to stall endlessly. I'll add a unit test instead.

Reply:
I've opened #2797 to track the need for unit tests more generally. I will work on that separately.